homeassistant/helpers/network.py

```diff
@@ -263,8 +265,14 @@ def _get_external_url(
     require_current_request: bool = False,
     require_ssl: bool = False,
     require_standard_port: bool = False,
+    require_cloud: bool = False,
 ) -> str:
     """Get external URL of this instance."""
+    if require_cloud:
+        if not allow_cloud:
```

Comment:
I think it's weird to take both options into consideration. If cloud is required we should just get the cloud url I think, and not take any other option into consideration.